tests/jballerina-unit-test/src/test/resources/test-src/expressions/let/let-expression-test.bal

```diff
@@ -362,6 +362,25 @@ function testLetWithClass() {
     assert(6, foo2.m);
 }
 
+function testLetWithBLangXMLTextLiteral() {
+    xml x = xml `<Books>${xml `<Book>"${let int y = 5 in y}"</Book>`}</Books>`;
+    assert(xml `<Book>"5"</Book>`, (x[0]/*));
+}
+
+function testLetExprWithBLangXMLCommentLiteral(){
+    xml x = xml `<Books>${xml `<Book><!--${let int y = 5 in y}--></Book>`}</Books>`;
+    assert(xml `<Book><!--5--></Book>`, (x[0]/*));
+
+    int i = 1;
+    xml z = xml `<Books>${xml `<Book><!--${let function () returns int y = (() => i) in y()}--></Book>`}</Books>`;
```

Comment:
We need to add such tests for the other two kinds also. Or we won't be testing rewriting the concatExpr.